### DISCUSSION TITLE
ResponseHeaders for no route and all routes

### DISCUSSION THREAD

**Discussion by app2eds** [dotnet/yarp#1928] [Q&A] [answered]
I have some response headers that must be added to all responses (mandated by our security policies).

Is it possible, through configuration, to add response headers when a matching route is not found? 

Is it possible, through configuration, to add a specific response header to ALL matching routes without having to add the ResponseHeader to each individual route?

I can create middleware to do this for all responses but would prefer configuration if possible.

**Comment by Tratcher** [comment, marked as answer]
> Is it possible, through configuration, to add a specific response header to ALL matching routes without having to add the ResponseHeader to each individual route?

This is not currently possible, that would have to be done via middleware. https://github.com/microsoft/reverse-proxy/issues/164 is the closest proposal I've seen for this.

> Is it possible, through configuration, to add response headers when a matching route is not found?

This should be possible in 2.0 with a wildcard route and either https://github.com/microsoft/reverse-proxy/pull/1923 or https://github.com/microsoft/reverse-proxy/issues/1893
